src/SourceBuild/tarball/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/DotNetWatchTests.cs

```diff
@@ -50,11 +56,13 @@ void processConfigCallback(Process process)
                 }
                 else if (e.Data?.Contains(expectedString) ?? false)
                 {
-                    outputChanged = true;
                     OutputHelper.WriteLine("Successfully re-ran program after code change.");
-                    ExecuteHelper.ExecuteProcessValidateExitCode("kill", $"-s TERM {process.Id}", OutputHelper);
+                    process.Kill(true);
+                    tcs.TrySetResult(true);
```

Comment:
What's the value in using a TaskCompletionSource here versus just setting a boolean at this point similar to the original code?